specification/scvmm/ScVmm.Management/models.tsp

```diff
@@ -640,6 +643,36 @@ model OsProfileForVmInstance {
   /** Gets os version. */
   @visibility("read")
   osVersion?: string;
+
+  /** Gets or sets the domain name. */
+  @added(Versions.v2024_06_01)
+  domainName?: string;
+
+  /** Gets or sets the domain username. */
+  @added(Versions.v2024_06_01)
+  domainUsername?: string;
+
+  /** Password of the domain the VM has to join. */
+  @visibility("create", "update")
+  @secret
+  @added(Versions.v2024_06_01)
+  domainPassword?: string;
+
+  /** Gets or sets the workgroup. */
+  @added(Versions.v2024_06_01)
+  workgroup?: string;
+
+  /** Gets or sets the product key.Input format xxxxx-xxxxx-xxxxx-xxxxx-xxxxx */
+  @added(Versions.v2024_06_01)
+  productKey?: string;
+
+  /** Gets or sets the index value of the timezone. */
+  @added(Versions.v2024_06_01)
+  timezone?: int32;
+
+  /** Get or sets the commands to be run once at the time of creation separated by semicolons. */
+  @added(Versions.v2024_06_01)
+  runOnceCommands?: string;
```

Comment:
If these commands are run only during creation, the values will be persisted inside object forever? What would be the behavior during update operation?

Reply:
Answer to your first question Yes during get operations we will be able to the see the commands.
Second question - These added properties are inside the Osprofile which is only present in VirtualMachineInstanceProperties and not in the VirtualMachineInstanceUpdateProperties.Thus the user wont be able to change any properties under the os profile.